lightning/src/ln/peer_handler.rs

```diff
@@ -1401,10 +1401,17 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 
 		// Need an Init as first message
 		if let wire::Message::Init(msg) = message {
-			if msg.features.requires_unknown_bits() {
-				log_debug!(self.logger, "Peer features required unknown version bits");
+			let our_features = self.init_features(&their_node_id);
+			if msg.features.requires_unknown_bits_from(&our_features) {
+				log_debug!(self.logger, "Peer requires features unknown to us");
 				return Err(PeerHandleError { }.into());
 			}
+
+			if our_features.requires_unknown_bits_from(&msg.features) {
+				log_debug!(self.logger, "We require features unknown to our peer");
+				return Err(PeerHandleError { }.into());
+			}
```

Comment:
Seeing this hit in `full_stack.rs` fuzz test because our peer only has the static remote key required while we have many other required features. @TheBlueMatt How would prefer we handle this without making this test too fragile when we add required features to our handlers?

Reply:
`ChannelTypeFeatures::from_init` is causing problems here. So I think we also need to modify the `open_channel` messages accordingly.

```
ERROR [lightning::ln::channelmanager : /Users/jkczyz/src/rust-lightning/lightning/src/ln/channelmanager.rs, 6488] Only static_remote_key is supported for non-negotiated channel types
```

Reply:
Heh, well, I'm getting somewhere and getting accustomed to editing the comment / hex. Now I'm seeing:

```
ERROR [lightning::ln::channelmanager : /Users/jkczyz/src/rust-lightning/lightning/src/ln/channelmanager.rs, 6488] Peer is signaling upfront_shutdown but we don't get any script. Use 0-length script to opt-out
```

I suppose the simplest fix would be to clear that bit in the peer's `init` message since `ChannelManager` doesn't require it.

Reply:
Done and also need in `accept_channel`.